### PR TITLE
Smartstart fixes

### PR DESCRIPTION
**Pull request by verbman**
* Tax and benefit system evolution.
* Impacted periods: all
* Impacted areas: 
 - `openfisca_aotearoa/variables/acts/income_tax/family_scheme/`
 - `openfisca_aotearoa/variables/acts/social_security/`
 - `openfisca_aotearoa/variables/demographics/`
 - `openfisca_aotearoa/variables/other/`
 - `openfisca_aotearoa/variables/regulation/social_security/childcare_assistance/`
 - `openfisca_aotearoa/variables/regulation/student_allowance/`
* Details:
 - * Renamed variables:
  - `social_security__eligible_for_childcare_subsidy` to `social_security_regulation__eligible_for_childcare_subsidy`
  - `sole_parent__family_has_child_under_age_limit` to `sole_parent_support__family_has_child_under_age_limit`
  - `family_has_resident_child_under_5_not_in_school` to `social_security_regulation__family_has_resident_child_under_5_not_in_school`
  - `family_has_resident_child_aged_5_who_will_be_enrolled_in_school` to `social_security_regulation__family_has_resident_child_aged_5_who_will_be_enrolled_in_school`
  - `family_has_child_eligible_for_disability_allowance_child_under_6` to `social_security_regulation__family_has_child_eligible_for_disability_allowance_child_under_6`
 - * New variables:
  - `family_scheme__full_time_earner`
  - `hours_per_week_employed`
  - `early_childcare_hours_participation_per_week`
  - `student_allowance__partner_or_person_receiving_certain_allowances`
 - * New parameters:
  - `entitlements.social_security.family_scheme.hours_per_week_threshold`
  - `entitlements.social_security.family_scheme.hours_per_week_threshold_with_partner`
  - `entitlements.social_security.childcare_subsidy.minimum_hours_in_childcare`
 - * Added `social_security_regulation` to structure.json

- - - -

These changes:

- Impact the OpenFisca-Aotearoa public API (for instance renaming or removing a variable)
- Add a new feature (for instance adding a variable)
- Fix or improve an already existing calculation.
- Change non-functional parts of this repository (for instance editing the README)

### CODE REVIEW

File: openfisca_aotearoa/tests/income_tax/family_scheme/best_start.yaml [hunk 1]
```diff
@@ -141,12 +162,19 @@
...
     best_start__entitlement:
       2018-07: [260, 0, 0, 0]
+
 - name: Tests paid parental leave entitlement value with baby due date after 1st July 2018
   period: 2018
   absolute_error_margin: 0
   persons:
     - id: "Papatūānuku"
       date_of_birth: 1979-02-01
+      social_security__is_ordinarily_resident_in_new_zealand:
```
Comment (Br3nda) [inline openfisca_aotearoa/tests/income_tax/family_scheme/best_start.yaml]: I can see this added to the tests, but I can't fun where you've added this to the PPL code.

How about a test for the scenrior where `social_security__is_ordinarily_resident_in_new_zealand` is false.

Also a reference. Where in the PPL act does it require the social security act definition of being resident in NZ.

I'm reviewing on a small screen of my phone so sorry if I've missed it.

Comment (verbman) [reply openfisca_aotearoa/tests/income_tax/family_scheme/best_start.yaml]: It appears the tests were named incorrectly, they are nothing to paid parental leave - but rather best start entitlement (the file is named correctly).
I have now also added a test for false as suggested.

File: openfisca_aotearoa/tests/income_tax/family_scheme/family_tax_credit.yaml [hunk 2]
```diff
@@ -8,10 +8,13 @@
...
         2018: 20000
       family_scheme__family_tax_credit_income_under_threshold: true
+      social_security__is_ordinarily_resident_in_new_zealand: true
+      hours_per_week_employed: 40
     - id: "Rakinui"
       date_of_birth: 1996-01-01
       family_scheme__assessable_income:
         2018: 20000
+      social_security__is_ordinarily_resident_in_new_zealand: true
     - id: "Tāne"
```
Comment (Br3nda) [inline openfisca_aotearoa/tests/income_tax/family_scheme/family_tax_credit.yaml]: is the child required to reside in NZ too?

File: openfisca_aotearoa/tests/social_security/childcare_subsidy.yaml [hunk 3]
```diff
@@ -20,16 +22,16 @@
       children:
         - "Child"
   output_variables:
-    family_has_resident_child_under_5_not_in_school:
+    social_security_regulation__family_has_resident_child_under_5_not_in_school:
       2018-08:
         - true  #  Whanau
-    family_has_resident_child_aged_5_who_will_be_enrolled_in_school:
+    social_security_regulation__family_has_resident_child_aged_5_who_will_be_enrolled_in_school:
```
Comment (Br3nda) [inline openfisca_aotearoa/tests/social_security/childcare_subsidy.yaml]: gosh the variables are getting long. I'm thinking we look into getting names spacing into the variables in OF core this year.

File: openfisca_aotearoa/tests/income_tax/family_scheme/best_start.yaml (hunk 1 above)
Comment (Br3nda) [reply openfisca_aotearoa/tests/income_tax/family_scheme/best_start.yaml]: So the Income Tax Act is using the Social Security Act definition of ordinarily resident in NZ?
Is there a reference URL? I thought Income Tax act has it's own residence definition.

**Comment by Br3nda** [comment]
Merge whenever you're ready. You'll need to stick around to watch that it deploys correctly.

File: openfisca_aotearoa/tests/income_tax/family_scheme/best_start.yaml (hunk 1 above)
Comment (verbman) [reply openfisca_aotearoa/tests/income_tax/family_scheme/best_start.yaml]: I've got it wrong :-(. It was instructed that we use this social security definition, but after a re-investigation I'll need to revert these particular changes and instead advise the smartstart team to utilise the `income_tax__residence` variable.

**Pull request merged**